controller/domain.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/que-go"
+	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/net/context"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/pkg/httphelper"
+	"github.com/flynn/flynn/pkg/postgres"
+	"github.com/flynn/flynn/pkg/tlscert"
+)
+
+type DomainMigrationRepo struct {
+	db *postgres.DB
+}
+
+func NewDomainMigrationRepo(db *postgres.DB) *DomainMigrationRepo {
+	return &DomainMigrationRepo{db: db}
+}
+
+func (repo *DomainMigrationRepo) Add(dm *ct.DomainMigration) error {
+	tx, err := repo.db.Begin()
+	if err != nil {
+		return err
+	}
+	if err := tx.QueryRow("INSERT INTO domain_migrations (old_domain, domain, old_tls_cert, tls_cert) VALUES ($1, $2, $3, $4) RETURNING migration_id, created_at", dm.OldDomain, dm.Domain, dm.OldTLSCert, dm.TLSCert).Scan(&dm.ID, &dm.CreatedAt); err != nil {
```

Comment:
This should use a [prepared query](https://github.com/flynn/flynn/blob/v20151013.0/controller/schema/queries.go).